great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -145,6 +149,11 @@ def __init__(
         if variables is None:
             variables = {}
 
+        if data_context:
+            self._usage_statistics_handler = data_context.usage_statistics_handler
+        else:
+            self._usage_statistics_handler = None
+
```

Comment:
Thoughts on passing the handler itself to the constructor as opposed to relying on the data context property? If we want to eventually get away from the context as sort of a catch-all, we could supply the handler from the context and propagate it downstream.

We've done something similar for the ProfilerStore on the context.

Reply:
I like passing the handler, but we might want to instead create a registry where we can retrieve the same handler instance based on data_context_id of the data_context requesting run of the RBP.

Reply:
We have GREAT-613 (a spike to determine the viability of a registry) in our backlog. I think for now I'll pass the handler and revisit this down the line?

I love the idea but don't want scope to increase if that's okay with you.